ziprofs.py

```diff
@@ -1,116 +1,84 @@
 #!/usr/bin/env python3
 from __future__ import print_function, absolute_import, division
 
-import logging
+from functools import lru_cache
 
 from os.path import realpath
-from sys import argv, exit
-from threading import Lock
 
 import argparse
 import errno
 import logging
 import os
 import zipfile
 import stat
+from threading import Lock
+from typing import Optional
 
-from fusepy import FUSE, FuseOSError, Operations, LoggingMixIn, S_IFDIR
-from collections import OrderedDict
+try:
+    from fuse import FUSE, FuseOSError, Operations, LoggingMixIn, S_IFDIR
+except ImportError:
+    # ubuntu renamed package in repository
+    from fusepy import FUSE, FuseOSError, Operations, LoggingMixIn, S_IFDIR
 
+from collections import OrderedDict
 
-class CachedZipFile(object):
-    MAX_SEEK_READ = 1 << 24
 
-    def __init__(self, path):
-        self.cache = {}
-        self.zf = zipfile.ZipFile(path)
-
-    def read(self, subpath, size, offset):
-        if subpath not in self.cache:
-            self.cache[subpath] = (0, self.zf.open(subpath))
-        pos, f = self.cache[subpath]
-        if f.seekable():
-            pos = f.seek(offset)
-            buf = f.read(size)
-        else:
-            if offset < pos:
-                pos = 0
-                f.close()
-                f = self.zf.open(subpath)
-            else:
-                offset -= pos
-            while offset > 0:
-                read_len = min(self.MAX_SEEK_READ, offset)
-                buf = f.read(read_len)
-                pos += len(buf)
-                offset -= read_len
-            buf = f.read(size)
-
-        pos += len(buf)
-        self.cache[subpath] = (pos, f)
-        return buf
-
-    def close(self):
-        for k, v in self.cache.items():
-            v[1].close()
-        self.zf.close()
-
-    # pass through methods
-    def getinfo(self, subpath):
-        return self.zf.getinfo(subpath)
-
-    def infolist(self):
-        return self.zf.infolist()
+@lru_cache(maxsize=2048)
```

Comment:
When I do not use this, whole dir listing is noticeably slower.
I don't feel like leaving it to kernel fs caching is good solution.

I probably could get mtime in account and write own caching, but not atm.

Also.. I've weird idea..
What If we give mtime as argument to this function? It will cache only version with the same mtime.
What do you think?